ssl/bio_ssl.c

```diff
@@ -357,7 +357,10 @@ static long ssl_ctrl(BIO *b, int cmd, long num, void *ptr)
             break;
         case SSL_ERROR_WANT_CONNECT:
             BIO_set_flags(b, BIO_FLAGS_IO_SPECIAL | BIO_FLAGS_SHOULD_RETRY);
-            BIO_set_retry_reason(b, BIO_get_retry_reason(next));
+            if (next != NULL)
+                BIO_set_retry_reason(b, BIO_get_retry_reason(next));
+            else
+                BIO_set_retry_reason(b, BIO_RR_CONNECT);
```

Comment:
IMO it is impossible to get next == NULL here.

Reply:
Excuse me, but could you please share your opinion about? I would like to understand this

Reply:
Thank you very much for your explanations. Really, you are right and here cann`t be next == NULL. Now I think that ossl_assert() is also not needed here. I close the request if you don't mind.